### PR TITLE
Consistency and other improvements/fixes in docs

### PR DESCRIPTION
**Pull request by jbloom22**

### CODE REVIEW

File: python/hail/dataset.py [hunk 1]
```diff
@@ -133,7 +133,7 @@ def is_dosage(self):
 
     @handle_py4j
     def file_version(self):
-        """File version of dataset.
+        """File version of VariantDataset.
```
Comment (tpoterba) [inline python/hail/dataset.py]: incorrect capitalization, should be 'variant dataset'

Comment (jbloom22) [inline python/hail/dataset.py]: We tend to use HailContext, VariantDataset, KeyTable, etc in the top-line method doc, with Hail context, variant dataset, and key table below (except when the class is a link). I'm fine changing all to the later. Thoughts?

Comment (tpoterba) [inline python/hail/dataset.py]: I think we should only use the full capitalized name when we link to it

**Comment by jbloom22** [comment]
@tpoterba I made those changes and a bunch more.

**Pull request merged**